### DISCUSSION TITLE
Copilot not working/Not showing suggestions?

### DISCUSSION THREAD

**Discussion by Kevin-Kwan** [community/community#6973] [Copilot Conversations]
Not sure if my situation is the same with others, but I'm using Copilot in IntelliJ. And yes, I have removed the other channel and installed the Copilot extension from the marketplace for IntelliJ.
The problem is that when I am writing my Java classes, I am no longer getting the "class header" or even a "class body" suggestion implementations from Copilot unlike when I tested Copilot yesterday.
I am not getting any suggestions at all while writing code as well.
I have checked that I have relogged into GitHub.
Tried reinstalling Copilot, relogging in, and restarting my IDE multiple times.

**Comment by effortprogrammer** [comment]
@Kevin-Kwan Even though I'm using vs code, when I coded with java, I have the same problem. Yesterday, it worked fantastically. But today, for some reason, nothing showed. Not only java but also other language is in trouble.

**Reply by mathminds** [reply]
I am also getting no response for even something as simple as 

```python
# add two numbers
def 
```

**Comment by TimePPT** [comment]
The same problem I met about two hours ago.

**Reply by Kevin-Kwan** [reply]
Did you ever manage to resolve the issue by now? Or are you still experiencing the issue?

**Reply by TimePPT** [reply]
Still so...

**Comment by parthmakawana** [comment]
I've just got access to copilot technical preview, I am trying to test it using VS Code right now, but its not showing any suggestions.

**Comment by mlgthatsme** [comment]
In github copilot logs in visual studio code I get `Unhandled status from server:,504,{"error":{"message":"request timeout"}}` and no suggestions like a lot of people here

**Comment by mlgthatsme** [comment]
So from what I can tell it seems to be account specific? Is there throttling going on?

**Reply by parthmakawana** [reply]
Yes, it seems so. For me, it works sometimes, and after some time, it starts giving request timeout error.

**Comment by TarunavBA** [comment]
Hey @Kevin-Kwan for me too GitHub copilot is not detected. At all. At first I was so excited that I have got access to copilot. I got the access yesterday 3:47 am

**Reply by N3RDIUM** [reply]
It's working perfectly fine for me, tho!

**Comment by DetachHead** [comment]
i had the same issue, turns out it was because i didn't accept the telemetry prompt (i thought it was optional), when i opened `idea.log` it was full of these warnings:
```
WARN - efaultCopilotCompletionService - fetchCompletions: telemetry terms were not accepted 
```

i think if it's not optional it should show you a message after you click no that says it's mandatory

**Reply by jansorg** [reply]
Thanks! We're considering improvements to the messaging.

**Comment by Martmists-GH** [comment]
Similar issue seems to occur whenever my network changes, which happens pretty often on my laptop:

```
2021-10-31 02:41:12,450 [  52538]   WARN - efaultCopilotCompletionService - fetchCompletions: GitHub Copilot token secret not defined or invalid: null 
2021-10-31 02:41:15,248 [  55336]   WARN - efaultCopilotCompletionService - fetchCompletions: GitHub Copilot token secret not defined or invalid: null 
```

**Reply by davecheney** [reply]
@jansorg this may be a question for you.

**Reply by jansorg** [reply]
@martmists-gh Thanks for reporting! We're tracking this internally now.

**Reply by jansorg** [reply]
@martmists-gh Can you tell if this problem is persisting for a longer time or if it's only temporary? Are completions shown again when your network is online again?

**Reply by Martmists-GH** [reply]
I always only try when the network is online. It just seems that when I get home, the token is suddenly `null`, but the plugin still seems to think I'm logged in. I always need to re-login to activate it. 
One possibility I can think of is that I use IntelliJ at university and PyCharm at home, and I have it linked to both. Could this be related to the problem?

**Reply by jansorg** [reply]
Thanks! There's a bug in the current version, which isn't properly re-fetching a new token when the IDE is launched, which currently re-prompts for a new login. This bug will be fixed with the next update. It seems like it may be the possible cause of the problems you're seeing. Please let us know if the next update isn't fixing the problem for you.

**Reply by Martmists-GH** [reply]
With the new update that released today the issue still persists. 

1. Launch IntelliJ
2. Log in to copilot
3. Exit IntelliJ
4. Launch IntelliJ
5. No longer logged in to copilot

**Reply by jansorg** [reply]
That's most likely another known issue, https://github.com/github/feedback/discussions/8074
We'll fix this with one of the next updates.

**Comment by netgio** [comment]
I had the same issue, but the cause is different... for me it is untrusted CA certs.    
[ERROR] [default] [2021-11-18T11:14:39.014Z] GitHub Copilot could not connect to server. Extension activation failed: "unable to get local issuer certificate"

Our environment uses ZScaler as an outbound proxy, which requires us to set up trusted org-issued CA certs.  A lot of hoop jumping ensues to find where each app/process loads it's trusted CA certs from and work out how to add the additional certs. 

I resolved a similar issue with Github signin for the CoPilot extension by adding the root AND intermediate certs to Windows Trusted Root CA list and used the win-ca VSCode extension to load them into the Node.js CA context.  However, even with GH sign-in working, CoPilot itself does not "see" the org-issued CA certs.   

Where does the CoPilot extension load its CA cert list from?   Any suggestions welcome!!

**Comment by ghost** [comment]
### I Have Same Issue Copilet is Active And Icon show me nothing wrong but nothing in there.

I did:
re install & not working, remove extensions folder after uninstall copilet & login again but nothing, uninstall vscode & remove all extension folder & re install again but nothing change, copilet `output` in vs code is empty no issue no world no anything ...

after install icon pop up need login & ok, then need waitlist & ok, then need access & ok, after that restart vs code icon show & active but nothing at all any body solve this problem ? 

check all not working problem but ... & solve this issue even network issue checked & no issue in there no blocking fire wall or like that ... test network change but ... test with proxy but ...


Target Device:

os : `windows 11 pro 64 bit - build : 22000.318`
vscode version : `1.62.3`


### The big problem is that there is no problem !!!!

icon :
![copilet icon](https://user-images.githubusercontent.com/69055273/143426447-9dd661dd-d9d2-4c2b-bf2e-57da19ea7f9d.png)

click in icon :
![copilet icon active](https://user-images.githubusercontent.com/69055273/143426480-94630bce-4e07-49bf-9dbb-f35fa7ec7b1d.png)

output :
![outlog](https://user-images.githubusercontent.com/69055273/143426494-8ca76bfc-2317-41c9-bc20-c2777f3712dc.png)

**Comment by vulbsti** [comment]
i was finally able to figure out disable and enable copilot. and add an exception to windows firewall for all requests from VScode. Also you have to agree to its telemetry terms after enabling it. it should probably show up in notifications after u add exception to firewall. 
worked like a charm after that.

**Comment by kinivi** [comment]
Same on VSCode on the macOS. Just stopped working

**Comment by M-Gonzalo** [comment]
Same problem here. Up until some point yesterday it was working fine. Now I don't see any suggestions. CTRL + INTRO works though.
I'm using vscode for Linux and working on JavaScript

**Reply by davecheney** [reply]
I'm sorry you're having problems. Can you please paste the output from View -> Output; GitHub Copilot?

**Reply by M-Gonzalo** [reply]
@davecheney 
When I just type out, there is no output whatsoever.
On Ctrl + intro, I see the context being sent and some posible solutions.

> [INFO] [solutions] [2021-11-29T10:25:48.876Z] prompt: // Language: javascript
> 
> ...context...
> 
> [INFO] [default] [2021-11-29T10:25:48.883Z] [fetchCompletions] engine https://copilot-proxy.githubusercontent.com/v1/engines/copilot-codex
> [INFO] [default] [2021-11-29T10:25:50.006Z] request.response: [https://copilot-proxy.githubusercontent.com/v1/engines/copilot-codex/completions] took 1119 ms
> [INFO] [streamChoices] [2021-11-29T10:25:51.816Z] solution 4 returned. finish reason: [null] finishOffset: [357] completionId: [{cmpl-49Y8zM87DcCp4lHcPyYeorLEBwWIx}] created: [{1638181549}]
> [INFO] [solutions] [2021-11-29T10:25:51.866Z] Open Copilot completion: [ ...suggestion... ]
> [INFO] [streamChoices] [2021-11-29T10:25:52.473Z] solution 9 returned. finish reason: [null] finishOffset: [428] completionId: [{cmpl-49Y8zM87DcCp4lHcPyYeorLEBwWIx}] created: [{1638181549}]
> [INFO] [solutions] [2021-11-29T10:25:52.490Z] Open Copilot completion: [ ...suggestion... ]
> [INFO] [streamChoices] [2021-11-29T10:25:52.558Z] solution 5 returned. finish reason: [null] finishOffset: [455] completionId: [{cmpl-49Y8zM87DcCp4lHcPyYeorLEBwWIx}] created: [{1638181549}]
> [INFO] [solutions] [2021-11-29T10:25:52.573Z] Open Copilot completion: [ ...suggestion... ]
> [INFO] [streamChoices] [2021-11-29T10:25:52.980Z] solution 3 returned. finish reason: [null] finishOffset: [501] completionId: [{cmpl-49Y8zM87DcCp4lHcPyYeorLEBwWIx}] created: [{1638181549}]
> [INFO] [solutions] [2021-11-29T10:25:52.990Z] Open Copilot completion: [ ...suggestion... ]
> [INFO] [streamChoices] [2021-11-29T10:25:53.191Z] solution 8 returned. finish reason: [null] finishOffset: [554] completionId: [{cmpl-49Y8zM87DcCp4lHcPyYeorLEBwWIx}] created: [{1638181549}]
> [INFO] [solutions] [2021-11-29T10:25:53.206Z] Open Copilot completion: [ ...suggestion... ]
> [INFO] [streamChoices] [2021-11-29T10:25:53.307Z] solution 0 returned. finish reason: [null] finishOffset: [557] completionId: [{cmpl-49Y8zM87DcCp4lHcPyYeorLEBwWIx}] created: [{1638181549}]
> [INFO] [solutions] [2021-11-29T10:25:53.320Z] Open Copilot completion: [ ...suggestion... ]
> [INFO] [streamChoices] [2021-11-29T10:25:53.348Z] solution 1 returned. finish reason: [null] finishOffset: [573] completionId: [{cmpl-49Y8zM87DcCp4lHcPyYeorLEBwWIx}] created: [{1638181549}]
> [INFO] [solutions] [2021-11-29T10:25:53.362Z] Open Copilot completion: [ ...suggestion... ]
> [INFO] [streamChoices] [2021-11-29T10:25:53.601Z] solution 2 returned. finish reason: [null] finishOffset: [706] completionId: [{cmpl-49Y8zM87DcCp4lHcPyYeorLEBwWIx}] created: [{1638181549}]
> [INFO] [solutions] [2021-11-29T10:25:53.630Z] Open Copilot completion: [ ...suggestion... ]
> [INFO] [streamChoices] [2021-11-29T10:25:55.913Z] solution 7 returned. finish reason: ["stop"] finishOffset: [816] completionId: [{cmpl-49Y8zM87DcCp4lHcPyYeorLEBwWIx}] created: [{1638181549}]
> [INFO] [solutions] [2021-11-29T10:25:55.932Z] Open Copilot completion: [ ...suggestion... ]
> [INFO] [streamChoices] [2021-11-29T10:25:56.129Z] solution 6 returned. finish reason: ["stop"] finishOffset: [undefined] completionId: [{cmpl-49Y8zM87DcCp4lHcPyYeorLEBwWIx}] created: [{1638181549}]
> [INFO] [solutions] [2021-11-29T10:25:56.144Z] Open Copilot completion: [ ...suggestion... ]
>

**Reply by M-Gonzalo** [reply]
Good news! Apparently Copilot depends on autosuggestions being active. Only the intellisense suggestions were actually getting in the way of Copilot and productivity in general so I deactivated them

**Reply by kinivi** [reply]
@M-Gonzalo how did you do that?

**Reply by M-Gonzalo** [reply]
> @M-Gonzalo how did you do that?

On the settings. There are a lot of options, none of which seemed to matter because the damn suggestions kept appearing on every single keystroke. What I finally did was set the delay to some ridiculously high value. I believe it was like 10.000 milliseconds. Now I can code in peace 😁

**Reply by N3RDIUM** [reply]
If you want both to work at the same time, you have to press the `esc` key when the intellisense pops up.

**Comment by mozgbrasil** [comment]
Good morning my friends

It's not working here either

Thank you in advance to everyone who can contribute.

![Captura de tela de 2021-11-30 11-01-09](https://user-images.githubusercontent.com/17202780/144061514-29dfe4ca-96f0-4809-9218-d05d9909b453.png)

**Comment by Marmik2003** [comment]
Same for me too, copiilot is not working on pycharm pro (macOS monterey).
I tried: 
1. Relogin to copilot (Worked 1 in 25 times)
2. Reinstalled the copilot (no difference either)

**Reply by laureanorp** [reply]
Same OS/IDE, been experiencing this since two weeks ago or so.

**Reply by dinigo** [reply]
Same IDE and context here. Worked like charm before XMAss
![image](https://user-images.githubusercontent.com/3038962/148231206-de028c8b-73e0-48fd-b6cf-64ec3b4fdceb.png)

**Reply by N3RDIUM** [reply]
I guess this problem is due to too much load on the copilot servers.

**Comment by josh-degraw** [comment]
I've had copilot working perfectly for a week or two now in VSCode, and all of a sudden it stopped offering suggestions for me now. Hoping someone can guide me to some sort of resolution 🙏🏻

**Comment by M-Gonzalo** [comment]
I'm getting a 504 error all of the sudden on every try. HTML, CSS, python, etc...

```
[INFO] [ghostText] [2022-03-02T04:58:42.031Z] Breaking, no choices
[INFO] [default] [2022-03-02T05:02:36.951Z] [fetchCompletions] engine https://copilot-proxy.githubusercontent.com/v1/engines/copilot-codex
[INFO] [default] [2022-03-02T05:02:47.157Z] request.response: [https://copilot-proxy.githubusercontent.com/v1/engines/copilot-codex/completions] took 10205 ms
[ERROR] [fetch] [2022-03-02T05:02:47.177Z] Unhandled status from server:,504,{"error":{"message":"request timeout"}}

[INFO] [ghostText] [2022-03-02T05:02:47.178Z] Breaking, no choices
```

I tried restarting the IDE (VS Code) but nothing changed

**Reply by windwang** [reply]
Same for me too

**Reply by nhokmoon** [reply]
same with me
yesterday it worked fine

**Reply by nhokmoon** [reply]
and now it like you 
![image](https://user-images.githubusercontent.com/88638603/156301311-0120e6ea-4132-4137-8c3a-a8398a86ed32.png)

**Reply by Vishnu053** [reply]
I'm getting the same output as @nhokmoon since last hour. Worked perfectly till then. I tried re-launching vscode and re-installing, still the same.

**Reply by Vishnu053** [reply]
It is working for me now

**Reply by M-Gonzalo** [reply]
@Vishnu053 Same! Working now

**Reply by N3RDIUM** [reply]
I guess this problem is due to too much load on the copilot servers.

**Comment by kens-visuals** [comment]
I've just got the link, installed it, but it doesn't show any suggestions. However, when I read further in the instructions there's an option to generate multiple suggestions by pressing `Ctrl + Enter` and it opens another tab with already generated suggestions. So the overall functionality works fine, it just doesn't show suggestions while coding 🤷🏻‍♂️ 👨🏻‍💻

**Comment by Maximus1285** [comment]
I'm experiencing the same issue. Copilot is not making suggestions anymore. I'm using VSCode on Mac OS Monterey

**Comment by devidw** [comment]
Had the same issue a couple of times, simply no suggestions. In all cases, after a period of time (one to a few hours), things work properly again (without changing anything on the setup).

**Comment by GuilloSGit** [comment]
I made a clean installation of windows, and apparently I broke it.
Copilot is no longer working (1 month till today).

**Comment by grumber** [comment]
Same thing, here. Four days of enjoying copilot; now, nothing. And I was getting used to it...

**Reply by davecheney** [reply]
Can you please paste the relevant parts of View -> Output; select _GitHub Copilot_ from the pulldown.

**Reply by grumber** [reply]
It's back! I checked the output and there was nothing there. Then, after a while VSCode asked for Github authorization again. I authorized it, restarted it, and Copilot is (thankfully) back. Like when Socrates warned that we would lose our memory because of the advent of writing, I will no doubt un-learn how to code by myself after copilot... Awesome (though scary) stuff! Congratulations
